TFNetworkRecLayer.py

```diff
@@ -4770,6 +4773,20 @@ def get_dep_layers(self):
       ls.extend(self.explicit_search_sources)
     return ls
 
+  def post_process_final_rec_vars_outputs(self, rec_vars_outputs, seq_len):
+    """
+    :param dict[str,tf.Tensor] rec_vars_outputs:
+    :param tf.Tensor seq_len: shape (batch,)
+    :rtype: dict[str,tf.Tensor]
+    """
+    if self.length_normalization:
+      assert "choice_scores" in rec_vars_outputs
+      # Finalize length normalization. During search we keep an extra factor t (recurrent time step) for efficiency
+      # reasons (see self.get_output()). Remove it here.
+      num_time_steps = tf.reduce_max(seq_len) + 1  # + 1 to include sequence end
```

Comment:
Note that `+1` is not correct in all cases. You can configure whether the seq/RecLayer includes EOS or not.